### PR TITLE
chore/studio: Adds upgrade prompts to logging dashboard

### PR DESCRIPTION
**Pull request by Ziinc**
<details>
<summary>Previous implementation</summary>

This PR adds updated log prompts to the dashboard, customized for each tier. 
It also introduces the `useProjectSubscription` hook, for easily querying a project's subscription tier.
Upgrade button links to the billing page

image

</details>

This PR has been re-implemented to use the project subscription api, allowing customized upgrade prompts based on the user's input.

It also uses the `subscription.tier.supabase_prod_id` as the tier reference, instead of `subscription.tier.key`

Demos:

https://user-images.githubusercontent.com/22714384/165377573-ea199062-93c6-4c71-a618-38b8616cff6d.mov

Screenshot 2022-04-26 at 9 09 21 PM

Screenshot 2022-04-26 at 9 09 07 PM

Screenshot 2022-04-26 at 9 08 27 PM

Pre-merge checklist:
- [x] blocked by https://github.com/supabase/infrastructure/pull/4702

### CODE REVIEW

**Comment by MildTomato** [comment]
@phamhieu Can you take a look at this? 
This PR is using stripe/subscription to get subscriptions details, can @Ziinc get that subscription info from a store instead to save doing an additional call to fly api.

File: studio/hooks/misc/useProjectSubscription.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    current_period_end: number
+    current_period_start: number
+  }
+}
+
+interface Handlers {
+  refresh: () => void
+}
+
+const useProjectSubscription = (projectRef: string): [Data | null, Handlers] => {
```
Comment (MildTomato) [inline studio/hooks/misc/useProjectSubscription.tsx:24]: seems weird we need a hook for this.
already tagged Hieu in main PR comments about it

**Comment by MildTomato** [comment]
Everything is fine though other than that. 
Already spoke to you about changing this Logging UI around so I imagine this prompt might end up moving, but we should still get this out 😄

**Comment by Ziinc** [comment]
we can do something similar to [ProjectContentStore](https://github.com/supabase/supabase/blob/master/studio/stores/content/ProjectContentStore.ts) to chuck it in the store, which is also exposed [as a hook too](https://github.com/supabase/supabase/blob/1f5b3edc9f862d9f16883b766c8af273b2207a74/studio/stores/projectContentStore.ts#L16).

End result with swr is the same, with exception that swr handles revalidations automatically. Either way, fetched data is cached globally.

No issues with moving the prompt around.

**Comment by Ziinc** [comment]
Also added in timestamp filter checking
image

File: studio/hooks/misc/useProjectSubscription.tsx (hunk 1 above)
Comment (phamhieu) [reply studio/hooks/misc/useProjectSubscription.tsx:25]: It's ok for now to use `useProjectSubscription`. As we haven't pulled project tier info on `/projects` yet

However as we haven't landed https://github.com/supabase/infrastructure/pull/4702. This PR has to wait for that

**Comment by MildTomato** [comment]
This probably needs redoing now

**Comment by Ziinc** [comment]
Reimplemented, see PR description.

File: studio/components/interfaces/Settings/Logs/Logs.UpgradePrompt.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import { Button } from '@supabase/ui'
```
Comment (joshenlim) [inline studio/components/interfaces/Settings/Logs/Logs.UpgradePrompt.tsx:1]: small issue but just have the file named as `UpgradePrompt`?

File: studio/components/interfaces/Settings/Logs/Logs.constants.ts [hunk 3]
```diff
@@ -465,3 +465,9 @@ export const EXPLORER_DATEPICKER_HELPERS: DatetimeHelper[] = [
 
 export const getDefaultHelper = (helpers: DatetimeHelper[]) =>
   helpers.find((helper) => helper.default) || helpers[0]
+
+export const TIER_QUERY_LIMITS = {
+  FREE: { text: '1 day', value: 1, unit: 'day' },
+  PRO: { text: '7 day', value: 7, unit: 'day' },
```
Comment (joshenlim) [inline studio/components/interfaces/Settings/Logs/Logs.constants.ts:471]: `text` plural - 7 days, 3 months

**Review by joshenlim — APPROVED** [review]
just really small changes but lgtm
also, can you pull from master then push once more? so that the zone-www-dot-com check will be removed (we're moving the marketing site outside of this repo)

**Pull request merged**